src/react/posts.tsx

```diff
@@ -1,35 +1,36 @@
 import axios from 'axios';
 import { useState } from 'react';
-import { useSelector } from 'react-redux';
-import crossicon from '../images/Crossicon.png';
-import Post from "../components/post.js";
+import { UseAppSelector } from '../react/Hook';
+import {Post} from "../components/post";
+const crossicon = require ('../images/Crossicon.png');
 
-function Posts() {
-  const userId = useSelector((state) => state.auth.userId)
+const Posts: React.FC = () => {
+  const userId = UseAppSelector((state) => state.auth.userId)
   const [textarea, setTextarea] = useState('')
-  let [isOpen, setIsOpen] = useState(false) 
+  const [isOpen, setIsOpen] = useState(false) 
   const redirect = () => setIsOpen(prevState => !prevState)
```

Comment:
not a good name for a function. This redirect is not redirecting anywhere instead it is tooling a IsOpen state variable. Consider working on naming variables and functions so that it is clear of what the variable or function does.